flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -133,8 +133,8 @@
 		key("kubernetes.cluster-id")
 		.stringType()
 		.noDefaultValue()
-		.withDescription("The cluster id used for identifying the unique flink cluster. If it's not set, " +
-			"the client will generate a random UUID name.");
+		.withDescription("The cluster-id, which should be no more than 45 characters, is used for identifying " +
```

Comment:
I am not sure why do we only allow `45` characters here? Do you mean to leave some possible prefix/suffix for the future?

Reply:
Yes. The external Service has a suffix in its name now, and I used to file another ticket [FLINK-17032](https://issues.apache.org/jira/browse/FLINK-17032) to unify the naming convention of all the K8s components we created now, so I think it would be better to leave room for that.